internal/telemetry/telemetry.go

```diff
@@ -0,0 +1,41 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+// Package telemetry implements a client for sending telemetry information to
+// Datadog regarding usage of an APM library such as tracing or profiling.
+package telemetry
+
+// ProductChange enqueues an app-product-change event that signals a product has been turned on/off.
+// the caller can also specify additional configuration changes (e.g. profiler config info),
+// which will be sent via the app-client-configuration-change event
+func (c *Client) ProductChange(namespace Namespace, enabled bool, configuration []Configuration) {
```

Comment:
can this also toggling products being enabled or disabled then? It wouldn't hurt to describe how `enabled` is used in the godoc.